### PR TITLE
refactor(poolmanager): GetPoolDenoms method on PoolI for SQS

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Add GetPoolDenoms on PoolI interface in the poolmanager. Provide implementations for each pool.

This is useful for SQS.

### CODE REVIEW

**Comment by p0mvn** [comment]
@Mergifyio refresh

File: x/poolmanager/types/pool.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package types
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	proto "github.com/cosmos/gogoproto/proto"
+	proto "github.com/gogo/protobuf/proto"
```
Comment (mattverse) [inline x/poolmanager/types/pool.go:5]: Just curious, what's the reason we're changing this from cosmos' gogo proto to the original one?

File: x/cosmwasmpool/model/store_model.go [hunk 2]
```diff
@@ -96,3 +96,8 @@ func (p CosmWasmPool) GetStoreModel() poolmanagertypes.PoolI {
 func (p CosmWasmPool) SetWasmKeeper(wasmKeeper types.WasmKeeper) {
 	panic("CosmWasmPool.SetWasmKeeeper not implemented")
 }
+
+// GetPoolDenoms implements types.PoolI.
+func (p *CosmWasmPool) GetPoolDenoms(ctx sdk.Context) []string {
+	panic("CosmWasmPool.GetPoolDenoms not implemented")
```
Comment (mattverse) [inline x/cosmwasmpool/model/store_model.go:102]: Can we have it Return empty string for now?
Concerned that we might call this API from higher level (pool manager side or even worse somehow called in begin or end blocker), leaving us to panic unexpectedly

File: x/poolmanager/types/pool.go [hunk 3]
```diff
@@ -23,6 +23,9 @@ type PoolI interface {
 	// Returns whether the pool has swaps enabled at the moment
 	IsActive(ctx sdk.Context) bool
 
+	// GetPoolDenoms returns the pool's denoms.
+	GetPoolDenoms(sdk.Context) []string
```
Comment (mattverse) [inline x/poolmanager/types/pool.go:27]: Can we add simple table test cases for this?

**Review by mattverse — COMMENTED** [review]
Left some comments, please take a look!

File: x/cosmwasmpool/model/store_model.go (hunk 2 above)
Comment (czarcas7ic) [reply x/cosmwasmpool/model/store_model.go:102]: This is okay, we panic on every store model here as we do not intend to use any of them directly here

File: x/poolmanager/types/pool.go (hunk 1 above)
Comment (czarcas7ic) [reply x/poolmanager/types/pool.go:5]: Correct, we need to switch this back, good catch (was likely merge conflict)

**Review by czarcas7ic — APPROVED** [review]

**Comment by czarcas7ic** [comment]
Mentioning here that I had issues with this method for cwpools on mainet

**Comment by p0mvn** [comment]
> Mentioning here that I had issues with this method for cwpools on mainet

What was the solution to the issue?

**Comment by czarcas7ic** [comment]
@p0mvn I just changed the GetPools method on the cwpool side to return the keeper as well. It's in my draft PR for implementing the route query

**Comment by p0mvn** [comment]
> @p0mvn I just changed the GetPools method on the cwpool side to return the keeper as well. It's in my draft PR for implementing the route query

OK, thanks for getting back. I will then merge this, and let you fix the issue in your PR.

**Comment by p0mvn** [comment]
OK tests added

File: x/poolmanager/types/pool.go (hunk 1 above)
Comment (p0mvn) [reply x/poolmanager/types/pool.go]: done

**Pull request merged**